src/components/Modal/ModalWrapper.tsx

```diff
@@ -121,46 +118,74 @@ const ModalWrapper: React.FC<ModalWrapperProps> = ({
   className,
   style,
   padding,
+  children,
 }) => {
   /**
    * Renders the appopriate variant based on the availability of a
    * panel prop and its position.
    * @param modalProps
    */
-  const ModalProxy = (modalProps: {
-    maxContentHeight: string
-    minContentHeight: string
-    header: ReactNode
-    footer: ReactNode
-    body: ReactNode
-  }) => {
+  const ModalProxy = (maxContentHeight: string, minContentHeight: string) => {
     if (!panel) {
-      return <Modal {...modalProps} />
+      return (
+        <Modal
+          padding={padding}
+          maxContentHeight={maxContentHeight}
+          minContentHeight={minContentHeight}>
+          {children}
+        </Modal>
+      )
     }
-    switch (panel.position) {
+    switch (panel) {
       case "left":
-        return <LeftPanelModal {...modalProps} panelContent={panel.content} />
+        return (
+          <LeftPanelModal
+            padding={padding}
+            maxContentHeight={maxContentHeight}
+            minContentHeight={minContentHeight}>
+            {children}
+          </LeftPanelModal>
+        )
       case "right":
-        return <RightPanelModal {...modalProps} panelContent={panel.content} />
+        return (
+          <RightPanelModal
+            padding={padding}
+            maxContentHeight={maxContentHeight}
+            minContentHeight={minContentHeight}>
+            {children}
+          </RightPanelModal>
+        )
       case "bottom":
-        return <BottomPanelModal {...modalProps} panelContent={panel.content} />
+        return (
+          <BottomPanelModal
+            padding={padding}
+            maxContentHeight={maxContentHeight}
+            minContentHeight={minContentHeight}>
+            {children}
+          </BottomPanelModal>
+        )
       default:
-        return <Modal {...modalProps} />
+        return (
+          <Modal
+            padding={padding}
+            maxContentHeight={maxContentHeight}
+            minContentHeight={minContentHeight}>
+            {children}
+          </Modal>
+        )
     }
   }
 
-  const [showFooterBorder, setShowFooterBorder] = useState(false)
-  const setBodyRef = (el: HTMLElement | null) => {
-    if (el && el.scrollHeight > el.clientHeight) {
-      setShowFooterBorder(true)
-    }
-  }
+  // const [showFooterBorder, setShowFooterBorder] = useState(false)
```

Comment:
I can see that the border on overflow is not working anymore, however that is required in the specs...